telemetry/prometheus/metrics.go

```diff
@@ -28,9 +30,9 @@ type metrics struct {
 	cfg *Config
 
 	gaugeVecs     map[string]*prometheus.GaugeVec
-	counterVecs   map[string]*prometheus.CounterVec
+	counterVecs   sync.Map
 	histogramVecs map[string]*prometheus.HistogramVec
-	summaryVecs   map[string]*prometheus.SummaryVec
+	summaryVecs   sync.Map
```

Comment:
why are we not doing the same sync protection for gaugeVecs and histogramVecs? I think we should be consistent

Reply:
Yeap will do the same ones